sources/packages/backend/apps/api/src/route-controllers/program-year/_tests_/e2e/program-year.students.controller.getProgramYears.e2e-spec.ts

```diff
@@ -0,0 +1,157 @@
+  let appDataSource: DataSource;
+  beforeAll(async () => {
+    const { nestApplication, dataSource } = await createTestingAppModule();
+    appDataSource = dataSource;
+    app = nestApplication;
+  });
+
+  it("Should get the list of the available program years.", async () => {
+    // Arrange
+    const endpoint = "/students/program-year/options-list";
+    const studentToken = await getStudentToken(
+      FakeStudentUsersTypes.FakeStudentUserType1,
+    );
+
+    // Act/Assert
+    await request(app.getHttpServer())
+      .get(endpoint)
+      .auth(studentToken, BEARER_AUTH_TYPE)
+      .expect(HttpStatus.OK)
+      .then((response) => {
+        const programYearList = response.body;
+        expect(programYearList).toHaveLength(3);
+        expect(programYearList).toEqual(
+          expect.arrayContaining([
+            expect.objectContaining({
+              id: 1,
+              description:
+                "(2021-2022) - Study starting between August 01, 2021 and July 31, 2022",
+            }),
+            expect.objectContaining({
+              id: 2,
+              description:
+                "(2022-2023) - Study starting between August 01, 2022 and July 31, 2023",
+            }),
+            expect.objectContaining({
+              id: 3,
+              description:
+                "(2023-2024) - Study starting between August 01, 2023 and July 31, 2024",
+            }),
+          ]),
+        );
+      });
+  });
+
+  it("Should not include inactive program years in the list.", async () => {
+    // Arrange
+    const programYearRepo = appDataSource.getRepository(ProgramYear);
+    const programYear = await programYearRepo.findOneBy({ id: 2 });
+    programYear.active = false;
+    await programYearRepo.save(programYear);
+    const endpoint = "/students/program-year/options-list";
+    const studentToken = await getStudentToken(
+      FakeStudentUsersTypes.FakeStudentUserType1,
+    );
+
+    // Act/Assert
+    await request(app.getHttpServer())
+      .get(endpoint)
+      .auth(studentToken, BEARER_AUTH_TYPE)
+      .expect(HttpStatus.OK)
+      .then((response) => {
+        const programYearList = response.body;
+        expect(programYearList).toHaveLength(2);
+        expect(programYearList).toEqual(
+          expect.arrayContaining([
+            expect.objectContaining({
+              id: 1,
+              description:
+                "(2021-2022) - Study starting between August 01, 2021 and July 31, 2022",
+            }),
+            expect.objectContaining({
+              id: 3,
+              description:
+                "(2023-2024) - Study starting between August 01, 2023 and July 31, 2024",
+            }),
+          ]),
+        );
+      });
+
+    // Revert
+    programYear.active = true;
+    await programYearRepo.save(programYear);
+  });
+
+  it("Should get an active program year by id.", async () => {
+    // Arrange
+    const endpoint = "/students/program-year/1/active";
```

Comment:
I would recommend that the test create the program year as required and executed the test on that record.
We may rely on some DB seeding in some cases, but I believe that program year would not be one of them.

Reply:
Ok. I can setup them on beforeAll and remove them on afterAll, right? But also it may interfere in other tests that might be running at the same time, no?

Reply:
First, you do not need to remove them, unless there is a need to. They will be created on the test DB only.
They can be created `beforeAll` to be shared as needed but if the tests will be changing them, in a way that could interfere with other tests in the same file, the specific test can create some program years records also as needed.